lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -95,8 +95,11 @@ sub _allocate_jobs ($self, $free_workers) {
         next if $checked_jobs{$j->{id}};
         next unless @{$j->{matching_workers}};
         my $tobescheduled = _to_be_scheduled($j, $scheduled_jobs);
+        if (!defined $tobescheduled) {
```

Comment:
correct me but I think `_to_be_scheduled_recurse` doesnt return undef. shouldnt that be just `!$tobescheduled`?